test/python/write_test_data.py

```diff
@@ -50,6 +52,13 @@ def _new_test_object():
     d.parameter3 = 12
     d.option_choice2 = InstantiatedTemplateChoice(d.parameter3)
     d.option_choice2.default_field = 707
+
+    # The next test is for the correct lookup of choice selector types. The 
+    # choice below is using an enum, whose values also exist in a different enum
+    d.choice_selector = SomeEnum.HAS_A
+    d.basic_choice = BasicChoice(d.choice_selector)
+    d.basic_choice.has_a = 5
```

Comment:
At some point this may become harder and harder to understand when a deserialization failure occurs because the test will only tell us that we have a bad string of bytes. In the future (later PRs) it would be great to investigate if we can decouple this into multiple test cases/godog features so that each `test` is a separate binary, which we can debug easier if a regression occurs.

Reply:
I will ignore this for now, but let's consider this for the future.